src/main/resources/com/google/api/codegen/ruby/test.snip

```diff
@@ -0,0 +1,173 @@
+          @# operations performed on Methods work on Procs. The differences are:
+          @# Proc implements: :yield, :lambda?, :binding
+          @# Method implements: :receiver, :name, :original_name, :owner, :unbind,
+          @#                    and :super_method
+          lambda { |*args| public_send(name_symbol, *args) }
+        end
+
+        # TODO(jcanizales): Instead of mock_client, pass a block that verifies
+        # the host string and that the 2nd argument isn't
+        # :this_channel_is_insecure.
+        {@gRPCClass}::Stub.stub(:new, mock_client) { yield(mock_client) }
+      end
+
+      @join testCase : testViewModel.testCases
+        @let methodView = testCase.methodView, \
+             methodName = methodView.name
+          describe "#{@methodName}" do
+            it "forwards calls to {@gRPCClass}::Stub" do
+              stub_grpc_client do |mock_client|
+                # TODO(jcanizales): Initialize enums with the second value, strings
+                # with "sample", arrays with 1 element, etc. Do the same with the
+                # mocked response.
+                # TODO(jcanizales): Instead of aliasing all used types,
+                # include Google::Cloud::Language::V1beta1 in the sample.
+                {@sampleCodeAliases(methodView)}
+
+                expected_request = {@testCase.requestTypeName}.new(
+                  {@initRequest(methodView.initCode)}
+                )
+
+                mocked_response = {@testCase.responseTypeName}.new
+                # TODO(jcanizales): testCase.mockResponse.initCode
+
+                mock_client.expect(
+                  :{@methodName},
+                  mocked_response,
+                  [expected_request, Hash],
+                )
+                @switch testCase.sampleResponseName.toString
+                @case "Empty"
+                  # No response verifier to declare.
+                @case "Response"
+
+                  process_response = ->(response) {
+                    @# Verification.
+                    response.must_equal mocked_response
+                  }
+                @case "Element"
+
+                  process_element = ->(element) {
+                    @# TODO(jcanizales): verify a mocked element.
+                  }
+                @case "ResultsAndMetadata"
+
+                  process_results = ->(results) {
+                    @# TODO(jcanizales): Verify mocked results.
+                  }
+
+                  process_metadata = ->(metadata) {
+                    @# TODO(jcanizales): Verify mocked metadata.
+                  }
+                @default
+
+                  $unhandledCase: {@methodView.sampleResponseName.toString}$
+                @end
+
+                {@sampleCodeCore(methodView)}
+              end
+            end
+          end
+          
+        @end
+      @end
+    @end    
+  end
+
+@end
+
+# The following were adapted from initcode.snip.
```

Comment:
Check the difference between what `{@initCode(apiMethod.initCode)}` outputs:

```ruby
shelf = Shelf.new
books = []
series_string = "foobar"
series_uuid = SeriesUuid.new
series_uuid.series_string = series_string
```

and what this outputs:

```ruby
shelf: Shelf.new,
books: [],
series_string: "foobar",
series_uuid: SeriesUuid.new(
  series_string: series_string
),
```